docs/explanations/default-options.rst

```diff
@@ -0,0 +1,64 @@
+.. SPDX-FileCopyrightText: 2023 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _explanations-default-options:
+
+*********************
+Default build options
+*********************
+
+Meson offers many `built-in options <https://mesonbuild.com/Builtin-options.html>`__,
+and in the vast majority of cases those have good defaults. There are a couple
+of cases however where ``meson-python`` either needs to or chooses to override
+those with its own defaults. To view what those are for the version of
```

Comment:
possible improvement:

We could split the paragraph here and also document why we might choose to override the defaults. I wrote about the reasoning in https://github.com/mesonbuild/meson-python/pull/325#issuecomment-1463902837.

We could also note that the options we do need to set are mostly for build/installation details that should not be relevant to the user. The only exception is the native file, but the user should be able to override the parts of it that make sense (IIRC everything but the Python interpreter).

Reply:
If we keep `meson install` I would much prefer if we would get rid of the `--prefix` and `--python.{plat,pure}lib` options and use the `$DESTDIR` environment variable pointing to somewhere in `/tmp` instead. This would have the additional benefit of putting the installation directory on tmpfs, with possible significant speedups when building large wheels.

Reply:
> We could split the paragraph here and also document why we might choose to override the defaults. I wrote about the reasoning in [#325 (comment)](https://github.com/mesonbuild/meson-python/pull/325#issuecomment-1463902837).

I checked it in more detail, and it's effectively the same as what I already have in this PR: _"because Meson defaults to values that are appropriate for development, while the main purpose of meson-python is to build release artifacts."_

Reply:
> If we keep `meson install` I would much prefer if we would get rid of the `--prefix` and `--python.{plat,pure}lib` options and use the `$DESTDIR` environment variable pointing to somewhere in `/tmp` instead.

I don't think this is even a question, is it :rofl:? Those options and `DESTDIR` serve two different purposes. We only set those options for the heuristics, so they can be removed now.

Regarding `DESTDIR`, we already set it to a subdirectory of `working_dir`, which currently when using the PEP 517 hooks is always set by `tempfile.TemporaryDirectory` (in `with_temp_working_dir`), so isn't that true already?

Reply:
The working dir is created inside the project directory, thus it is on the same filesystem as the project sources.

Reply:
Ah, you're right :sweat_smile:. IIRC I did that to make things work properly for packagers by default, where access to `/tmp` might be a bit tricky. We can consider changing the default, though.

Reply:
I'll leave this comment thread open, because it seems like useful info. It doesn't apply to this PR directly though; it's discussion of a potential future design change. If that materializes, the docs can simply be updated.